### ISSUE TITLE
Mi Band HRX

### ISSUE THREAD

**Issue by dheerajgattupalli** [Freeyourgadget/Gadgetbridge#837]
App unable to connect to Mi band hrx it is discovered and paired but not connecting  .... i was previously using Mifit and it connected so band is functional ... i uninstalled mi fit and came gadgetbridge .... mi band is not connecting

[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/1364561/gadgetbridge.log)
*In case of a bug, do not forget to attach logs!*

#### Your wearable device is:Mi band HRX
Firmware version v1.0.1.52

*Please specify model and firmware version if possible*

#### Your Gadgetbridge version is:0.21.3

*New issues about already solved/documented topics could be closed without further comments. Same for too generic or incomplete reports.*

**Comment by ashimokawa** [comment]
@dheerajgattupalli 
We were not aware of the device. Looks like a Mi Band 2, we will try to support that with the next release.

**Comment by dheerajgattupalli** [comment]
Yes its an exact copy of mi band 2 minus the heart rate sensor ....  Thanks for quick response.... Will be waiting for new release ....

**Comment by dheerajgattupalli** [comment]
Sorry about the close and add .... It was an accident...

**Comment by ashimokawa** [comment]
This got delayed because we wanted to merge two bigger things before (one of them was pebble background js which just got merged)

**Comment by prashanthpai** [comment]
I'm not sure if this helps but here it goes.
The firmware for Mi Band HRX Edition is named `Mili_pro_i.fw` in the Mi Fit apk (3.1.0).

```sh
$ grep "HRX" Mili_pro_i.fw
Binary file Mili_pro_i.fw matches
$ grep "1.0.1.52" Mili_pro_i.fw
Binary file Mili_pro_i.fw matches
```

**Comment by ashimokawa** [comment]
Sure that helps. Maybe _i stands for India (?)

**Comment by prashanthpai** [comment]
> Sure that helps. Maybe _i stands for India (?)

May be. AFAICT, the [Mi Band HRX Edition](http://www.mi.com/in/mi-band-hrx/) is only released in India. It is exactly like Mi Band 2 but without the heart rate sensor.

**Comment by ashimokawa** [comment]
@dheerajgattupalli
@prashanthpai 

I have blindly added Mi Band HRX support in master, if you can please test.

**Comment by dheerajgattupalli** [comment]
Hello i want to test but the build is not yet available on f-droid ..... where can i find the app?

**Comment by dheerajgattupalli** [comment]
[gadgetbridge[1].log](https://github.com/Freeyourgadget/Gadgetbridge/files/1415852/gadgetbridge.1.log)
[gadgetbridge(1)[1].log](https://github.com/Freeyourgadget/Gadgetbridge/files/1415854/gadgetbridge.1.1.log)

Hello I was unable wait till the build was released so went ahead built the source myself and turns out that the app is still not able to connect to my hrx band.... please check out the logs .....if your devices support addition is not yet completed sorry for disturbing you .....

**Comment by ashimokawa** [comment]
@dheerajgattupalli 
Did you pair through Gadgetbridge?
Did you uninstall Mifit?

Please try to unpair from android settings and then press the + button in the main activity and try to connect.

**Comment by prashanthpai** [comment]
I gave it a shot. After multiple attempts it does connect to HRX band. Most of the things (buttons) on the debug page doesn't work. Call notification works. I haven't had a chance to test further yet.

**Comment by dheerajgattupalli** [comment]
Yes after a few attempts it got connected ..... Notifications from app in debug are working.... Will verify further ....

**Comment by golijeevankumar** [comment]
How do u connected it from where u found the new version please update me

**Comment by danielegobbetti** [comment]
I deleted a few comments here. Please remember this is an **issue** tracker!

**Comment by golijeevankumar** [comment]
Any updates for mi hrx band

**Comment by ashimokawa** [comment]
Try it and report back
